client/client.go

```diff
@@ -98,6 +103,14 @@ func Configure(logger hclog.Logger, config interface{}) (schema.ClientMeta, diag
 		logger.Info("No subscriptions specified going to using all available ones", "subscriptions", subscriptions)
 	}
 
+	// validate subscriptions
+	for i, subscriptionId := range client.subscriptions {
+		if !validateSubscriptionId(subscriptionId) {
+			logger.Error("Invalid subscription id", "subscription_id", subscriptionId)
+			client.subscriptions = append(client.subscriptions[:i], client.subscriptions[i+1:]...)
```

Comment:
I believe you can run into trouble if you remove elements from a slice while also ranging over it. Here's a [playground demo](https://go.dev/play/p/ALCwNnaM9wL) to illustrate. What you can do, though, is this trick from the SliceTricks page: https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating (that's my `good` example in the playground)